docs/guides/serverless-shared-migration-guide.md

```diff
@@ -0,0 +1,266 @@
+---
+page_title: "Migration Guide: Transition out of Serverless Instances and Shared-tier clusters"
+---
+
+# Migration Guide: Transition out of Serverless Instances and Shared-tier clusters
+
+The goal of this guide is to help users transition from Serverless Instances and Shared-tier clusters (M2/M5) to Free, Flex or Dedicated Clusters. 
+
+Starting in March 2025, all Shared-tier clusters (in both `mongodbatlas_cluster` and `mongodbatlas_advanced_cluster`) will automatically convert to Flex clusters. Similarly, all Serverless instances (`mongodb_serverless_instance`) will be converted into Free/Flex/Dedicated clusters, [depending on your existing configuration](https://www.mongodb.com/docs/atlas/flex-migration/). 
+These conversions will result in configuration drift in Terraform. 
```

Comment:
not always right? shared tier migrates to flex but still works as expected

Reply:
I believe cases where configuration drift occurs (in March 2025 or whenever auto-migration is done) are when serverless is converted to either free or dedicated.
When serverless or shared tier get converted to flex, API endpoints keep working until January 2026

Reply:
changed a bit the phrasing in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2746/commits/b6944698fdb057950b42b5163edcda5449344094, also changed the date of shared tier autoconversion to January, which is what we have in the migrtion doc from the upstream team